content/oci/oci_test.go

```diff
@@ -2237,6 +2274,123 @@ func TestStore_PredecessorsAndDelete(t *testing.T) {
 	}
 }
 
+func TestStore_Untag(t *testing.T) {
+	content := []byte("test delete")
+	desc := ocispec.Descriptor{
+		MediaType: "test-delete",
+		Digest:    digest.FromBytes(content),
+		Size:      int64(len(content)),
+	}
+	ref := "latest"
+
+	tempDir := t.TempDir()
+	s, err := New(tempDir)
+	if err != nil {
+		t.Fatal("NewDeletableStore() error =", err)
+	}
+	ctx := context.Background()
+
+	err = s.Push(ctx, desc, bytes.NewReader(content))
+	if err != nil {
+		t.Errorf("Store.Push() error = %v, wantErr %v", err, false)
+	}
+
+	err = s.Tag(ctx, desc, ref)
+	if err != nil {
+		t.Errorf("error tagging descriptor error = %v, wantErr %v", err, false)
+	}
+
+	exists, err := s.Exists(ctx, desc)
+	if err != nil {
+		t.Fatal("Store.Exists() error =", err)
+	}
+	if !exists {
+		t.Errorf("Store.Exists() = %v, want %v", exists, true)
+	}
+
+	resolvedDescr, err := s.Resolve(ctx, ref)
+	if err != nil {
+		t.Errorf("error resolving descriptor error = %v, wantErr %v", err, false)
+	}
+
+	if !reflect.DeepEqual(resolvedDescr, desc) {
+		t.Errorf("Store.Resolve() = %v, want %v", resolvedDescr, desc)
+	}
+
+	err = s.Untag(ctx, ref)
+	if err != nil {
+		t.Errorf("Store.Untag() = %v, wantErr %v", err, nil)
+	}
+
+	_, err = s.Resolve(ctx, ref)
+	if err == nil {
+		t.Errorf("error resolving descriptor error = %v, wantErr %v", err, errdef.ErrNotFound)
+	}
+
+	exists, err = s.Exists(ctx, desc)
+	if err != nil {
+		t.Fatal("Store.Exists() error =", err)
+	}
+	if !exists {
+		t.Errorf("Store.Exists() = %v, want %v", exists, true)
+	}
+}
+
+func TestStore_UntagErrorPath(t *testing.T) {
+	content := []byte("test delete")
+	desc := ocispec.Descriptor{
+		MediaType: "test-delete",
+		Digest:    digest.FromBytes(content),
+		Size:      int64(len(content)),
+	}
+	ref := "latest"
+
+	tempDir := t.TempDir()
+	s, err := New(tempDir)
+	if err != nil {
+		t.Fatal("NewDeletableStore() error =", err)
+	}
+	ctx := context.Background()
+
+	err = s.Untag(ctx, "")
+	if err == nil {
+		t.Errorf("Store.Untag() error = %v, wantErr %v", err, errdef.ErrMissingReference)
```

Comment:
The error can potentially be of other type which is not nil and not `ErrMissingReference`, so it is better to test if the error type matches.

```suggestion
	if !errors.Is(err, errdef.ErrMissingReference) {
		t.Errorf("Store.Untag() error = %v, wantErr %v", err, errdef.ErrMissingReference)
```

This also applies to other similar places.